apps/cowswap-frontend/src/modules/trade/hooks/setupTradeState/useSetupTradeStateFromUrl.ts

```diff
@@ -16,36 +24,69 @@ import { useTradeState } from '../useTradeState'
 export function useSetupTradeStateFromUrl(): null {
   const params = useParams()
   const location = useLocation()
-  const stringifiedParams = JSON.stringify(params)
   const setState = useSetAtom(tradeStateFromUrlAtom)
   const { state } = useTradeState()
   const tradeStateRef = useRef(state)
   tradeStateRef.current = state
 
-  /**
-   * useEffect() runs after the render completes and useMemo() runs during rendering.
-   * In order to update tradeStateFromUrlAtom faster we use useMemo() here.
-   * We need this, because useSetupTradeState() depends on the atom value and needs it to be udpated ASAP.
-   */
-  useMemo(() => {
+  const { chainId, inputCurrencyId, outputCurrencyId, recipient, recipientAddress, targetChainId } = useMemo(() => {
+    const stringifiedParams = JSON.stringify(params)
```

Comment:
I believe the point of stringifying the params outside of the hook is to make the dependency stable.
The way it's used here is not very useful, as it stringifies here to parse on line 35/38